### PR TITLE
Add inference percentile details to driver perf

### PR DESCRIPTION
**Pull request by richagadgil**
Use flag --r or --runtimes with perf to export times to JSON

Updated perf output:

Batch size: 1
Rate: 898.852 inferences/sec
Total time: 1.11253ms (Min: 1.095ms, Max: 3.7384ms, Mean: 1.14045ms, Median: 1.11241ms)
Percentiles (90%, 95%, 99%): (1.12391ms, 1.13541ms, 1.22018ms)
Total instructions time: 2.40689ms
Overhead time: 0.0542455ms, -1.29436ms
Overhead: 5%, -116%

### CODE REVIEW

File: src/program.cpp [hunk 1]
```diff
@@ -845,6 +847,50 @@ double common_average(const std::vector<double>& v)
...
+    {
+        file << migraphx::to_pretty_json_string(val, 4);
+        file.close();
+        std::cout << "Runtimes dumped to " << file_name << std::endl;
+    }
+    else
+    {
+        std::cerr << "Failed to open file: " << file_name << std::endl;
+    }
+}
```
Comment (pfultz2) [inline src/program.cpp:892]: Remove the json dumping in this PR.

File: src/program.cpp [hunk 2]
```diff
@@ -879,8 +925,12 @@ void program::mark(const parameter_map& params, marker&& m)
...
 }
 
-void program::perf_report(
-    std::ostream& os, std::size_t n, parameter_map params, std::size_t batch, bool detailed) const
+void program::perf_report(std::ostream& os,
+                          std::size_t n,
+                          parameter_map params,
+                          std::size_t batch,
+                          bool detailed,
+                          bool runtimes) const
```
Comment (pfultz2) [inline src/program.cpp:933]: Remove this parameter.

File: src/program.cpp [hunk 3]
```diff
@@ -925,8 +951,14 @@ void program::perf_report(
     {
         overhead_vec.push_back(time<milliseconds>([&] { dry_run(params); }));
     }
-
     double total_time             = common_average(total_vec);
+    double min_time               = total_vec.front();
+    double max_time               = total_vec.back();
+    double mean_time              = mean(total_vec);
+    double median_time            = median(total_vec);
```
Comment (pfultz2) [inline src/program.cpp:958]: Isnt the median just `percentile(total_vec, 0.5)`?

**Review by shivadbhavsar — APPROVED** [review]

File: src/program.cpp (hunk 3 above)
Comment (shivadbhavsar) [reply src/program.cpp:958]: That would be slightly off for the even number of runs case I think

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**